### PR TITLE
OCPBUGS-45971: e2e: tuned degraded test fix

### PR DESCRIPTION
**Pull request by rbaturov**
* Addressed a corner case where nodes based on VMs would always log a warning without verifying if the profile was degraded. This incorrect behavior has been fixed to properly check the profile state before logging warnings.
* Modified isVM func to execute "systemd-detect-virt" command after chroot to rootfs. This will ensures accurate virtualization detection by running the command in the host's context instead of the container.

### CODE REVIEW

File: test/e2e/performanceprofile/functests/1_performance/performance.go
Comment (jmencak) [inline test/e2e/performanceprofile/functests/1_performance/performance.go:136]: Nit: I know the previous code had the same "issue", but perhaps using the "happy path" and inverting the condition might  make the code a bit more readable?

File: test/e2e/performanceprofile/functests/utils/infrastructure/vm.go
Comment (rbaturov) [reply test/e2e/performanceprofile/functests/utils/infrastructure/vm.go:13]: I should have used it.

File: test/e2e/performanceprofile/functests/1_performance/performance.go
Comment (rbaturov) [reply test/e2e/performanceprofile/functests/1_performance/performance.go:136]: If `isNotBasedOnVM` you mean?
Not sure I understood

Comment (jmencak) [reply test/e2e/performanceprofile/functests/1_performance/performance.go:136]: Actually, looking more closely, it looks fine as is. Sorry for the noise.

**Comment by rbaturov** [comment]
/cherry-pick release-4.18 release-4.17 release-4.16 release-4.15

File: test/e2e/performanceprofile/functests/1_performance/performance.go
Comment (jmencak) [inline test/e2e/performanceprofile/functests/1_performance/performance.go:135]: I guess it might help which Tuned profile is degraded (better logging), but not a show stopper.

**Review by jmencak — APPROVED** [review]
/approve
/lgtm
to give a chance to others to review before merging, feel free to unhold
/hold

**Pull request merged**
